File: Java OOP-Feb2020/src/j_polymorphism_Exercises/wildFarm/Animal.java
package j_polymorphism_Exercises.wildFarm;

public abstract class Animal {
    private String animalType;
    private String animalName;
    private Double animalWeight;
    private Integer foodEaten;

    public Animal(String animalType, String animalName, Double animalWeight) {
        this.animalType = animalType;
        this.animalName = animalName;
        this.animalWeight = animalWeight;
        this.foodEaten = 0;
    }

    public abstract void makeSound();

    public void eat(Food food) {
        this.foodEaten += food.getQuantity();
    }

    protected String getAnimalType() {
        return animalType;
    }

    protected String getAnimalName() {
        return animalName;
    }

    protected Double getAnimalWeight() {
        return animalWeight;
    }

    protected Integer getFoodEaten() {
        return foodEaten;
    }
}

File: Programming-for-QA-sep2023/21.Implementing-OOP-Hierarchy/WildFarm/Animal/Animal.cs
namespace WildFarm.Animal;
using WildFarm.Food;

public abstract class Animal
{
    public Animal(string name, double weight)
    {
        this.Name = name;
        this.Weight = weight;
    }

    public string Name { get; set; }
    public double Weight { get; set; }
    public int FoodEaten { get; set; }

    public abstract void AskForFood();

    public abstract void FeedAnimal(Food food);

}
